### PR TITLE
perf(cdk/a11y): tree-shake warnings in production

### PR DESCRIPTION
**Pull request by arturovt**

### CODE REVIEW

File: src/cdk/a11y/focus-trap/focus-trap.ts [hunk 1]
```diff
@@ -207,7 +209,10 @@ export class FocusTrap {
 
     if (redirectToElement) {
       // @breaking-change 8.0.0
-      if (redirectToElement.hasAttribute(`cdk-focus-initial`)) {
+      if (
+        (typeof ngDevMode === 'undefined' || ngDevMode) &&
```
Comment (jelbourn) [inline src/cdk/a11y/focus-trap/focus-trap.ts:213]: The warnings should be gated on dev mode, but I don't think this behavior here should be

Comment (devversion) [reply src/cdk/a11y/focus-trap/focus-trap.ts:213]: Unless I miss anything, this is just the warning being omitted in production mode. that seems reasonable to me.

**Review by devversion — APPROVED** [review]

File: src/cdk/a11y/focus-trap/focus-trap.ts (hunk 1 above)
Comment (jelbourn) [reply src/cdk/a11y/focus-trap/focus-trap.ts:213]: Nevermind, I misread the change

**Review by jelbourn — APPROVED** [review]
LGTM

**Pull request merged**